File: pos-frontend/src/app/models/upload-response.ts
export interface UploadError {
  rowNumber: number;
  data: string;
  message: string;
}

// ErrorEntry and UploadError are the same, so we can use just one
export type ErrorEntry = UploadError;

export interface UploadResponse {
  successfulEntries: any[];
  errors: UploadError[];
  totalRows: number;
  successCount: number;
  errorCount: number;
  failedEntries?: FailedEntry[];
}

export interface FailedEntry {
  row: number;
  data: string;
  errorMessage: string;
} 

File: pos/src/main/java/com/increff/model/UploadResponse.java
package com.increff.model;

import java.util.ArrayList;
import java.util.List;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class UploadResponse {
    private int totalRows;
    private int successCount;
    private int errorCount;
    private List<UploadError> errors = new ArrayList<>();
    private List<InventoryData> successfulEntries = new ArrayList<>();
} 